.github/workflows/ci.yml

```diff
@@ -56,6 +59,15 @@ jobs:
           sonatypeSnapshots: true
       - name: Build and test with Java ${{ matrix.java_version }}
         run: mvn --batch-mode --update-snapshots -P IT install
+
+      - name: Deploy GitHub Pages to snapshot
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./manual/target/generated-docs/
```

Comment:
This should not be hard-coded but taken from a workflow input (similar to release workflow)